src/test/java/io/cdap/plugin/successfactors/source/input/SuccessFactorsPartitionBuilderTest.java

```diff
@@ -0,0 +1,215 @@
+    Assert.assertEquals("Optimal load distribution count is not same",
+      (partitionList.size() - expectedExtraLoadCount), distributedLoadCount);
+  }
+
+  @Test
+  public void testUpdatedFetchRowCount() {
+    long availableRowCount = 123;
+    long fetchRowCount = 100;
+
+    List<SuccessFactorsInputSplit> partitionList = partitionBuilder.buildSplits(availableRowCount, fetchRowCount);
+
+    long expectedFetchSize = fetchRowCount;
+    long actualFetchSize =
+      partitionList.stream().collect(Collectors.summarizingLong(SuccessFactorsInputSplit::getBatchSize)).getSum();
+    Assert.assertEquals("Total record extraction count is not same", expectedFetchSize, actualFetchSize);
+  }
+
+  @Test
+  public void testPackageSizeOptimizationBasedOnSplitCount() {
+    long availableRowCount = 123;
+    long fetchRowCount = 100;
+
+    List<SuccessFactorsInputSplit> partitionList = partitionBuilder.buildSplits(availableRowCount, fetchRowCount);
+
+    long expectedPackageSize = fetchRowCount;
+    Assert.assertEquals("Package size is not optimized", expectedPackageSize,
+      partitionList.get(partitionList.size() - 1).getBatchSize());
+  }
+
+  @Test
+  public void testMaxPackageSizeOptimization() {
+    long availableRowCount = 2000;
+    long fetchRowCount = 1000;
+    int splitCount = 1;
+
+    List<SuccessFactorsInputSplit> partitionList = partitionBuilder.buildSplits(availableRowCount, fetchRowCount);
+
+    Assert.assertTrue("Package size is beyond the allowed optimized size.",
+      partitionList.stream()
+        .filter(successFactorsInputSplit -> successFactorsInputSplit.getBatchSize() == 1000)
+        .count() == splitCount);
+  }
+
+  @Test
+  public void testStartEndAndPackageSizeCompare() {
+    long availableRowCount = 9;
+    long fetchRowCount = 0;
+
+    List<SuccessFactorsInputSplit> partitionList = partitionBuilder.buildSplits(availableRowCount, fetchRowCount);
+
+    Assert.assertEquals("Start is not same for split 1", 1, partitionList.get(0).getStart());
+    Assert.assertEquals("End is not same for split 1", 9, partitionList.get(0).getEnd());
+    Assert.assertEquals("Package size is not same for split 1", 9, partitionList.get(0).getBatchSize());
+  }
+
+  @Test
+  public void testDefaultSplitCount() {
+    long availableRowCount = 9000;
+    long fetchRowCount = 0;
+
+    List<SuccessFactorsInputSplit> partitionList = partitionBuilder.buildSplits(availableRowCount, fetchRowCount);
+    Assert.assertEquals("Split count is not same", SuccessFactorsPartitionBuilder.DEFAULT_SPLIT_COUNT,
+                        partitionList.size());
+  }
+
+  @Test
+  public void testBatchSize() {
+    long availableRowCount = 9000;
+    long fetchRowCount = 0;
+
+    List<SuccessFactorsInputSplit> partitionList = partitionBuilder.buildSplits(availableRowCount, fetchRowCount);
+    Assert.assertEquals("Split count is not same", 8, partitionList.size());
```

Comment:
8 -> SuccessFactorsPartitionBuilder.DEFAULT_SPLIT_COUNT. 
We can remove the previous test case as it's testing the same thing.

Reply:
Replaced `8` with `SuccessFactorsPartitionBuilder.DEFAULT_SPLIT_COUNT` 046113e0e5c304322acae66d107442c4f5df1d63

Removed the previous test case. 95b017153f270805f657d526167195f67e551890